### PR TITLE
[DI2-102] Mobile header link fix

### PR DESCRIPTION
**Pull request by janjaap**
This PR fixes an issue on smaller screens where the search overlay would overlap the links in the site header, making them unclickable. The solution was to keep track of the overlay and to apply a specific CSS rule for that overlay so that, only when visible, should span the entire width of the viewport.

To ensure the functionality, integration tests have been added to the homepage spec.

### CODE REVIEW

File: src/app/components/SearchBar/SearchBar.tsx [hunk 1]
```diff
@@ -20,7 +20,13 @@ const Z_INDEX_OFFSET = 2 // Set a custom offset
 const StyledSearchBarToggle = styled(SearchBarToggle)`
```
Comment (jonkoops) [inline src/app/components/SearchBar/SearchBar.tsx:20]: You should be  able to add the type information here so you don't have to redefine it below.
```suggestion
const StyledSearchBarToggle = styled(SearchBarToggle)<{ searchIsVisible: boolean }>`
```

File: src/app/components/SearchBar/SearchBar.tsx [hunk 2]
```diff
@@ -20,7 +20,13 @@ const Z_INDEX_OFFSET = 2 // Set a custom offset
...
   z-index: 23;
   position: absolute;
-  left: 0;
+
+  ${({ searchIsVisible }: { searchIsVisible: boolean }) =>
```
Comment (jonkoops) [inline src/app/components/SearchBar/SearchBar.tsx:24]: So this could be removed
```suggestion
  ${({ searchIsVisible }) =>
```

File: src/app/components/SearchBar/SearchBar.tsx [hunk 3]
```diff
@@ -91,6 +98,10 @@ const SearchBar: FunctionComponent<SearchBarProps> = ({
     placeholder: placeHolder,
   }
 
+  const toggleWidth = (isOpen: boolean): void => {
```
Comment (jonkoops) [inline src/app/components/SearchBar/SearchBar.tsx:101]: No need to add `: void` here as the type can be inferred.`
```suggestion
  const toggleWidth = (isOpen: boolean) => {
```

File: src/header/components/HeaderSearch.tsx [hunk 4]
```diff
@@ -86,34 +86,37 @@ const HeaderSearch: FunctionComponent = () => {
...
-    }
 
-    // @ts-ignore
-    document.activeElement?.blur()
-  }
+      // @ts-ignore
+      document.activeElement?.blur()
+    },
+    [inputValue, selectedElement, searchBarFilterValue],
+  )
```
Comment (jonkoops) [inline src/header/components/HeaderSearch.tsx:119]: I believe you can get rid of this `@ts-ignore`:

```suggestion
   if (document.activeElement instanceof HTMLElement) {
     document.activeElement.blur()
   }
```

File: src/header/components/HeaderSearch.tsx [hunk 5]
```diff
@@ -86,34 +86,37 @@ const HeaderSearch: FunctionComponent = () => {
...
+      } else {
+        const queryString = {
+          term: inputValue.trim(),
+        }
+        const query = new URLSearchParams(
+          Object.entries(queryString).reduce(
+            (acc, [key, value]) => (value ? { ...acc, [key]: value } : acc),
+            {},
+          ),
+        )
```
Comment (jonkoops) [inline src/header/components/HeaderSearch.tsx:106]: Can't we put this hash directly into the constructor?
```suggestion
        const query = new URLSearchParams({
          term: inputValue.trim(),
        })
```

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: src/header/components/HeaderSearch.tsx (hunk 5 above)
Comment (janjaap) [reply src/header/components/HeaderSearch.tsx:106]: Possibly, but that code was already in the component.

Comment (jonkoops) [reply src/header/components/HeaderSearch.tsx:106]: Sure, but no harm leaving things better than they were before, no?

Comment (janjaap) [reply src/header/components/HeaderSearch.tsx:106]: Very true. But making a change like that requires adding tests, which there are none for this component. I'll create a different PR for that.

**Review by jonkoops — APPROVED** [review]

File: src/header/components/HeaderSearch.tsx (hunk 5 above)
Comment (jonkoops) [reply src/header/components/HeaderSearch.tsx]: I've simplified the code and tested it locally, looks like it was just redundant.

**Review by chrisvanmook — APPROVED** [review]
